pypeit/scripts/identify.py

```diff
@@ -86,55 +96,240 @@ def main(args):
-                                  PYP_SPEC=msarc.PYP_SPEC, lamps=','.join(lamps))
+                arccen, arc_maskslit = wavecal.extract_arcs(slitIDs=[slit_val])
+
+                # Launch the identify window
+                # TODO -- REMOVE THIS HACK
+                try:
+                    nonlinear_counts = msarc.detector.nonlinear_counts()
+                except AttributeError:
+                    nonlinear_counts = None
+                arcfitter = Identify.initialise(arccen, lamps, slits, slit=int(slit_val), par=par,
+                                                wv_calib_all=wv_calib_slit, wavelim=[args.wmin, args.wmax],
+                                                nonlinear_counts=nonlinear_counts,
+                                                pxtoler=args.pixtol, test=args.test, 
+                                                fwhm=args.fwhm,
+                                                sigdetect=args.sigdetect,
+                                                specname=spec.name,
+                                                y_log=not args.linear,
+                                                rescale_resid=args.rescale_resid)
+
+                # If testing, return now
+                if args.test:
+                    return arcfitter
+                final_fit = arcfitter.get_results()
+                fits_dicts.append(arcfitter._fitdict)
+                specdata.append(arccen[:,slit_val])
+                # Build here to avoid circular import
+                #  Note:  This needs to be duplicated in test_scripts.py
+                # Wavecalib (wanted when dealing with multiple detectors, eg. GMOS)
+                if 'WaveFit' in arcfitter._fitdict.keys():
+
+                    lines_pix_arr.append(arcfitter._fitdict['WaveFit']['pixel_fit'])
+                    lines_wav_arr.append(arcfitter._fitdict['WaveFit']['wave_fit'])
+                    lines_fit_ord.append(arcfitter._fitdict['WaveFit']['pypeitfit']['order'])
+                    if np.any(wv_calib.wv_fits):
+                        wv_calib.wv_fits[slit_val] = arcfitter._fitdict['WaveFit']
+                        wv_calib.wv_fits[slit_val].fwhm = measured_fwhms[slit_val]
+                    else:
+                        wv_fits_arr.append(arcfitter._fitdict['WaveFit'])
+                        wv_fits_arr[-1].fwhm = measured_fwhms[slit_val]
+                    waveCalib = wv_calib
+
+            
+                else:
+                    if np.any(wv_calib.wv_fits):
+                        wv_calib.wv_fits[slit_val] = None
+                    else: 
+                        wv_fits_arr.append('None')
+                    waveCalib = None
+                    custom_wav_q = ''
+                    while custom_wav_q != 'y' and custom_wav_q != 'n':
+                        custom_wav_q = input('No solution made! Do you want to input an estimated linear solution? y/[n]: ')
```

Comment:
For all of the various `input` values seen below, is it worth including the msgs.input() as well?